rust/arrow/src/buffer.rs

```diff
@@ -568,6 +568,25 @@ impl MutableBuffer {
     }
 }
 
+impl MutableBuffer {
+    /// Writes a byte slice to the underlying buffer and updates the `len`, i.e. the
+    /// number array elements in the builder.  Also, converts the `io::Result`
+    /// required by the `Write` trait to the Arrow `Result` type.
+    pub fn write_bytes(&mut self, bytes: &[u8], len_added: usize) -> Result<()> {
+        let write_result = self.write(bytes);
+        // `io::Result` has many options one of which we use, so pattern matching is
+        // overkill here
+        if write_result.is_err() {
+            Err(ArrowError::MemoryError(
```

Comment:
We have an `ArrowError::IoError`, or do you prefer `MemoryError` here?

Reply:
I'm fine with `IoError`, will change.